### DISCUSSION TITLE
Title to get attention

### DISCUSSION THREAD

**Discussion by CodeLieutenant** [phper-framework/phper#143] [General]
I'm so sorry for the title, I don't have any inspiration for the title about this topic that I would like to discuss 😄. First of all, I want to thank you for this awesome library. It's a real enjoyment to use, really easy to understand and extend. 

I'm the maintainer of the PHP extension for [ScyllaDB/Apache Cassandra](https://github.com/he4rt/scylladb-php-driver). For a long time, my buddy @DanielHe4rt and I wanted to rewrite the whole extension in Rust, for obvious reasons. It's an old codebase and contains some issues that aren't resolvable with the current state of the codebase (memory leaks, a mixture of C and C++, etc.). 

The rewrite has started on [V2 Branch](https://github.com/he4rt/scylladb-php-driver/tree/v2) since then, I found some missing parts needed for the full rewrite (e.g. Access to Zend Persistent Resources, Smart String API), so I forked the repo and made these changes needed. In the spirit of OpenSource, I wanted to announce these changes to you and if they align with your vision of the project we can merge them, if not, free feel to cherry-pick the stuff you like. Some changes are breaking (like replacing symtable API with str_* functions and dropping support for PHP < 8.1).

Changes can be found [HERE](https://github.com/dmalusev/phper).

**Comment by jmjoy** [comment]
Welcome to fork, I will continue to monitor your modifications.

But one thing that is inconsistent is that, I need to support `PHP7` because it involves my usage scenario.

**Comment by shijunti19** [comment]
@jmjoy I agree with your point of view that many systems are PHP 7.3 and they can never be upgraded, but they also hope to improve performance. rust is the only solution
